fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -1891,13 +1891,25 @@ void MultiVersionApi::loadEnvironmentVariableNetworkOptions() {
 		return;
 	}
 
+	FDBOptionParamType curParamType;
+	int64_t intParamVal;
 	for (auto option : FDBNetworkOptions::optionInfo) {
 		if (!option.second.hidden) {
 			std::string valueStr;
 			try {
 				if (platform::getEnvironmentVar(("FDB_NETWORK_OPTION_" + option.second.name).c_str(), valueStr)) {
+					curParamType = option.second.paramType;
 					for (auto value : parseOptionValues(valueStr)) {
-						Standalone<StringRef> currentValue = StringRef(value);
+						Standalone<StringRef> currentValue;
+						if (curParamType == Int) {
+							char* end;
+							intParamVal = strtoull(value.c_str(),&end,10);
+							ASSERT(!(*end) && intParamVal);
```

Comment:
If we continue using `strtoll`, we can't assert that `intParamVal` is non-zero because of the possibility that the user could set it to 0.

Reply:
I see. However, the code logic won't allow user to set it to 0, which could refer to:
`threadCount = extractIntOption(value, 1, 1024);`
Is that ok?

Reply:
That logic is specific to the client thread option, and so this would work for that option. However it is possible for other options to specify different bounds on the legal values.

I didn't think of this before, but we should also not ASSERT on bad input from a user. Rather, an ASSERT should be for conditions that we never expect to happen. If we get bad input, we should instead throw an error like `invalid_option_value`.